Rework of collection version select

jerabekjiri opened the pull request:
Issue: [AAH-907](https://issues.redhat.com/browse/AAH-907)

- Removed collection version select from collection-info.tsx 
- Used Patternfly Select instead of FormSelect for nicer styling

before:
![Screenshot from 2021-09-02 16-48-17](https://user-images.githubusercontent.com/19647757/131865630-5511dcf7-01b2-4835-8f9a-5550276ea6fa.png)
after:
![Screenshot from 2021-09-02 14-36-58](https://user-images.githubusercontent.com/19647757/131865680-9d801adf-a018-4c22-84f6-e2bdc5f7a092.png)



- Added modal for  full list of collections
![Screenshot from 2021-09-02 14-37-25](https://user-images.githubusercontent.com/19647757/131864893-281924d9-2923-477b-ba03-1078df63c06c.png)

Full content screen with removed version select inside install content and with new select component :)
![Screenshot from 2021-09-02 14-37-11](https://user-images.githubusercontent.com/19647757/131866114-8a087395-aeab-437a-bb2f-d03af244eaf7.png)

@ZitaNemeckova  please take a look :)

jerabekjiri commented:
Cc @sbuenafe-rh, also what's the limit for items (versions) shown in the dropdown? :) :arrow_down: 
![](https://user-images.githubusercontent.com/19647757/131865680-9d801adf-a018-4c22-84f6-e2bdc5f7a092.png)

sbuenafe-rh commented:
@jerabekjiri I would be display 10 items and then change the **Show more** to **View more** and the rest of the options are loaded.

jerabekjiri commented:
In the new commits:
### Select versions dropdown
- limit dropdown list to max 10 items
- Changed from `Show more` to `View more`
- if less than 10 items in dropdown list, hide `View more` button
![Screenshot from 2021-09-08 23-23-55](https://user-images.githubusercontent.com/19647757/132665868-1a92cfb3-dbfd-421d-af5b-c3fd0b30360b.png)
![Screenshot from 2021-09-08 23-24-37](https://user-images.githubusercontent.com/19647757/132666790-a16f01a1-e32c-43fb-ba49-0c372f878906.png)

### Versions modal 
- added top and bottom pagination
- default page_size for pagination is 20
- aligned top pagination with modal description
![Screenshot from 2021-09-09 02-09-33](https://user-images.githubusercontent.com/19647757/132665634-3a066b1a-92b4-4e68-9c31-695b8d5b1407.png)

- I noticed typo `keyords`, so changed it to `keywords`

ZitaNemeckova commented on src/components/headers/collection-header.tsx:
```diff
@@ -186,4 +315,33 @@ export class CollectionHeader extends React.Component<IProps> {
+    }));
+  };
+
+  private onClose(key: string) {
```
It would be better to remove this function and just call `this.setState({key: false})`. It will improve readability of the code. There's no need to re-enter same values to `setState`.

jerabekjiri replied:
I wanted to reuse that function, but knew it's not the nicest solution :laughing:

ZitaNemeckova commented on src/components/headers/collection-header.tsx:
```diff
@@ -186,4 +315,33 @@ export class CollectionHeader extends React.Component<IProps> {
+
+  private updatePaginationParams = ({ page, page_size }) => {
+    this.setState((prevState) => ({
+      ...prevState,
```
No need to re-enter same values to `this.setState`.

ZitaNemeckova commented on src/components/headers/collection-header.tsx:
```diff
@@ -186,4 +315,33 @@ export class CollectionHeader extends React.Component<IProps> {
+    this.setState((prevState) => ({
+      ...prevState,
+      modalPagination: {
+        page: page,
```
Nitpick but this can be just `page,` :)

ZitaNemeckova commented on src/components/headers/collection-header.tsx:
```diff
@@ -71,79 +111,168 @@ export class CollectionHeader extends React.Component<IProps> {
+                <ExternalLinkAltIcon />
+              </div>
+              {urlKeys.map((link) => {
+                const l = collection.latest_version.metadata[link.key];
```
A more descriptive name like `url`.

ZitaNemeckova reviewed (approved):
Tested in UI. LGTM 👍
